src/flb_config.c

```diff
@@ -161,6 +161,12 @@ struct flb_config *flb_config_init()
     config->http_port    = flb_strdup(FLB_CONFIG_HTTP_PORT);
 #endif
 
+    config->http_proxy = getenv("HTTP_PROXY");
+    if (strcmp(config->http_proxy, "")) {
```

Comment:
there is a chance that ```config->http_proxy``` becomes NULL and the strcmp will crash:

```
RETURN VALUE
       The getenv() function returns a pointer to the value in the environment, or NULL if there is no match.
```

in the same conditional make sure config->http_proxy is not NULL